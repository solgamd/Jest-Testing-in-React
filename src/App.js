import React, { Component } from 'react';
import './App.css';
import Question from './Question';

class App extends Component {
  constructor() {
    super();
    this.state = {
      count: 0
    }
  };

  makeIncrementer = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount
    }));


  increment = this.makeIncrementer(1);

  decrement = this.makeIncrementer(-1);

  render() {
    return (
      <>
        <div className="App">
          <h1>Count: {this.state.count}</h1>
          <button className="increment" onClick={this.increment}>Increment</button>
          <button className="decrement" onClick={this.decrement}>Decrement</button>
        </div>
        <Question />
        
      </>
    );
  }
}

export default App;
