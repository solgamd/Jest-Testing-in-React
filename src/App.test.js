import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  })
});

describe('App component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('h1').text();
    expect(text).toEqual('Count: 0');
  });

  it('increments count by 1 when increment button is clicked', () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find('button.increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('h1').text();
    expect(text).toEqual('Count: 1');
  });

  it('decrements count by 1 when decrement button is clicked', () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('button.decrement');
    decrementBtn.simulate('click');
    const text = wrapper.find('h1').text();
    expect(text).toEqual('Count: -1');
  })
});

