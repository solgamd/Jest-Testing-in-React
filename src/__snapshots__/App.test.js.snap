// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Count: 
    0
  </h1>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
